src/lepton/stream_interfaces.hh

```diff
@@ -60,6 +60,8 @@ struct ResizableByteBufferList {
 
 class VP8ComponentDecoder_SendToVirtualThread {
     ResizableByteBufferList vbuffers[Sirikata::MuxReader::MAX_STREAM_ID];
+    // Auto-release all buffers allocated in this class by saving them into:
+    std::vector<std::unique_ptr<ResizableByteBufferListNode>> allocated_buffers;
```

Comment:
Why is it done  like this?

Reply:
Every buffer allocated is added to this list. We clear this list between images (and it's automatically cleared by destructor), so buffers are auto-released at these points.

Overall, unique_ptr is a way to automatically release raw memory buffers used by an object.

Reply:
Why wasn't `vbuffers` used for that?

Reply:
Buffers are filled with data and pushed to the vbuffers. Then they are popped off vbuffers and returned as pair of raw pointers to be consumed by other class. Without going into details of this class, I just collect all the buffers ever allocated and release them at "safe points".

Alternatively, we may try to free previously allocated buffer when next buffer is started consuming. It would decrease decompression memory usage by 20-25%.

Reply:
Added as task MDS-6231.

While describing it, I recalled that PacketReader-descendant classes have alternative way to release buffers - setFree(). So I need to investigate how it interacts with allocated_buffers machinery - we may got double-freeing in some cases.